packages/material-ui/src/Select/SelectInput.js

```diff
@@ -8,35 +8,33 @@ import { refType } from '@material-ui/utils';
 import ownerDocument from '../utils/ownerDocument';
 import capitalize from '../utils/capitalize';
 import Menu from '../Menu/Menu';
-import { nativeSelectRootStyles, nativeSelectIconStyles } from '../NativeSelect/NativeSelectInput';
+import {
+  nativeSelectSelectStyles,
+  nativeSelectIconStyles,
+} from '../NativeSelect/NativeSelectInput';
 import { isFilled } from '../InputBase/utils';
 import experimentalStyled, { slotShouldForwardProp } from '../styles/experimentalStyled';
 import useForkRef from '../utils/useForkRef';
 import useControlled from '../utils/useControlled';
-import selectClasses, { getSelectUtilityClasses } from './selectClasses';
+import { getSelectUtilityClasses } from './selectClasses';
 
-const SelectRoot = experimentalStyled(
+const SelectSelect = experimentalStyled(
   'div',
   {},
   {
     name: 'MuiSelect',
-    slot: 'Root',
+    slot: 'Select',
     overridesResolver: (props, styles) => {
       const { styleProps } = props;
       return {
-        [`&.${selectClasses.select}`]: {
-          // TODO v5: remove `root` and `selectMenu`
-          ...styles.root,
-          ...styles.select,
-          ...styles.selectMenu,
-          ...styles[styleProps.variant],
-        },
+        ...styles.select,
+        ...styles[styleProps.variant],
       };
     },
   },
-)(nativeSelectRootStyles, {
+)(nativeSelectSelectStyles, {
   // Win specificity over the input base
-  [`&.${selectClasses.selectMenu}`]: {
+  [`&[role="button"]`]: {
```

Comment:
A self-reference might be clearer.

```suggestion
  [`&.${selectClasses.select}`]: {
```

Reply:
Fixed.